app/mailers/rgpd_mailer.rb

```diff
@@ -79,7 +79,7 @@ def rgpd_contact_error
       result = @send_in_blue.send_transac_email(email)
       Rails.logger.info "Email sent with id: #{result.inspect}"
     rescue SibApiV3Sdk::ApiError => e
-      Rails.logger.error "Exception when calling SMTPApi->send_transac_email: #{e.inspect} #{e.response_body.inspect}"
+      Sentry.capture_exception(e)
```

Comment:
if SENTRY_DSN is not defined then Sentry will not be initialized.

What will happen here ?

Reply:
Actually, I wasn't expected to use specific Sentry instruction. I expected to use the usual logger while sentry collect the logs transparently for me.

Reply:
> What will happen here ?

Nothing (cf https://github.com/getsentry/sentry-ruby#sentry-only-runs-when-sentry-dsn-is-set )

> Actually, I wasn't expected to use specific Sentry instruction. I expected to use the usual logger while sentry collect the logs transparently for me.

I only migrate the current behaviour here. The alternative it's not to rescue the error and let the job failed. In this case, it will have the same effect (there's no configured queue adapter, on this rails version it's the [async queue](https://api.rubyonrails.org/classes/ActiveJob/QueueAdapters/AsyncAdapter.html) by default, and by default there's no retry behaviour documented [here](https://edgeguides.rubyonrails.org/active_job_basics.html#retrying-or-discarding-failed-jobs)).

Btw you should setup a queue system which is more robust.

Reply:
ok great ! and great perspective too !

Another question, will these errors be logged in the logger also ? so we can keep track in the .log file ? or we can read the error in the console when running the app in interactive dev mode ?

Reply:
> Another question, will these errors be logged in the logger also ?

By default Sentry captures the exception raised thanks to a custom middleware, it won't change the behaviour of the app: if you configure it to log errors it will continue (and by default it is).

In this specific case however, the `rescue` block suppress the actual exception and it won't have logs (because .. there's no exception in the end to capture by Rails).

If you want to keep rails logs and sentry, we should remove the begin/rescue block and let the rails stack handle it.